neuralmagic/tests/test_nm-vllm_licenses.py

```diff
@@ -0,0 +1,146 @@
+    ("LICENSE.fastertransformer", r"Apache License\n\s*Version 2.0", ""),
+    ("LICENSE.gptq", "MIT License\n*.*turboderp", ""),
+    ("LICENSE.marlin", r"Apache License\n\s*Version 2.0", ""),
+    ("LICENSE.punica", r"Apache License\n\s*Version 2.0", ""),
+    ("LICENSE.squeezellm", "MIT License\n*.*SqueezeAILab", ""),
+    ("LICENSE.tensorrtllm", r"Apache License\n\s*Version 2.0", ""),
+    ("LICENSE.vllm", r"Apache License\n\s*Version 2.0", ""),
+    (
+        "LICENSE",
+        r".*NEURAL MAGIC COMMUNITY LICENSE AGREEMENT.*",
+        "",
+    ),
+    (
+        "METADATA",
+        ".*License: Neural Magic Community License",
+        "",
+    ),
+]
+
+
+class TestNMThirdPartyLicenseFiles:
+    """
+    These tests verify that the proper files for licensing purposes exist and
+    [generally] have the expected content.
+    """
+
+    @staticmethod
+    def _package_name_version() -> Tuple[str, str]:
+        """
+        provides the package name as pip understands it, and the package version
+        :return: (name, version)
+        """
+        site_package = vllm.__path__[0]
+        package_version = vllm.__version__
+        return site_package, package_version
+
+    @staticmethod
+    def _build_dist_info_path() -> Path:
+        """
+        builds a Path to the nm-vllm dist-info site package directory
+        """
+        (
+            package_name,
+            package_version,
+        ) = TestNMThirdPartyLicenseFiles._package_name_version()
+        package_name = package_name.replace("vllm", "nm_vllm")
+        return Path(f"{package_name}-{package_version}.dist-info")
+
+    def check_file_exists_and_content(self, file_name: str,
+                                      content_regex: str):
+        """
+        shared function to check license files
+        :param file_name: the file to check.
+        :param content_regex: the regular expression to search the file content
+        """
+        # since we want to ensure that the files are actually available to the
+        # user, this test function specifically looks for the files, rather than
+        # accessing dist-info metadata for the package
+        dist_info = self._build_dist_info_path()
+        file_path = dist_info / file_name
+
+        assert (file_path.exists()
+                ), f"failed to find the expected license info {file_path}"
+        license_text = file_path.read_text("utf-8")
+        assert re.search(content_regex, license_text), (
+            f"license file {file_path} does not have expected content matching "
+            f"{content_regex}")
+
+    @mark.parametrize(
+        ("file_name", "content_regex"),
+        [param(lf[0], lf[1], marks=lf[2]) for lf in THIRD_PARTY_LICENSE_FILES],
+    )
+    def test_common_license_file_presence_content(self, file_name: str,
+                                                  content_regex: str):
+        """
+        Check Neural Magic license files that are common to the community and
```

Comment:
nit: the comment seems out of place. i don't think we have an enterprise package.